crates/uv/src/commands/project/add.rs

```diff
@@ -183,57 +244,64 @@ pub(crate) async fn add(
     .await?;
 
     // Add the requirements to the `pyproject.toml`.
-    let existing = project.pyproject_toml();
-    let mut pyproject = PyProjectTomlMut::from_toml(existing)?;
+    let mut toml = match &dependency_destination {
+        DependencyDestination::Script(script) => {
+            PyProjectTomlMut::from_toml(&script.metadata.raw, DependencyTarget::Script)
+        }
+        DependencyDestination::Project(project) => {
+            let raw = project.pyproject_toml().raw.clone();
+            PyProjectTomlMut::from_toml(&raw, DependencyTarget::PyProjectToml)
+        }
+    }?;
     let mut edits = Vec::with_capacity(requirements.len());
     for mut requirement in requirements {
         // Add the specified extras.
         requirement.extras.extend(extras.iter().cloned());
         requirement.extras.sort_unstable();
         requirement.extras.dedup();
 
-        let (requirement, source) = if raw_sources {
-            // Use the PEP 508 requirement directly.
-            (pep508_rs::Requirement::from(requirement), None)
-        } else {
-            // Otherwise, try to construct the source.
-            let workspace = project
-                .workspace()
-                .packages()
-                .contains_key(&requirement.name);
-            let result = Source::from_requirement(
-                &requirement.name,
-                requirement.source.clone(),
-                workspace,
-                editable,
-                rev.clone(),
-                tag.clone(),
-                branch.clone(),
-            );
-
-            let source = match result {
-                Ok(source) => source,
-                Err(SourceError::UnresolvedReference(rev)) => {
-                    anyhow::bail!("Cannot resolve Git reference `{rev}` for requirement `{name}`. Specify the reference with one of `--tag`, `--branch`, or `--rev`, or use the `--raw-sources` flag.", name = requirement.name)
-                }
-                Err(err) => return Err(err.into()),
-            };
+        let (requirement, source) = match dependency_destination {
+            DependencyDestination::Script(_) => (pep508_rs::Requirement::from(requirement), None),
```

Comment:
These actually do support sources now.

Reply:
Added support for sources and a test for this